### PR TITLE
Migration of Self Signup Configuration

### PR DESCRIPTION
**Pull request by thisaltennakoon**
Fix https://github.com/wso2/api-manager/issues/361

### CODE REVIEW

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIConstants.java [hunk 1]
```diff
@@ -868,15 +867,10 @@ private Permissions() {
...
-    public static final String SELF_SIGN_UP_REG_DOMAIN_ELEM = "SignUpDomain";
+    //elements in the configuration file in the advance tenant configuration related to self signup
     public static final String SELF_SIGN_UP_REG_ROLES_ELEM = "SignUpRoles";
-    public static final String SELF_SIGN_UP_REG_ROLE_ELEM = "SignUpRole";
-    public static final String SELF_SIGN_UP_REG_USERNAME = "AdminUserName";
-    public static final String SELF_SIGN_UP_REG_PASSWORD = "AdminPassword";
-    public static final String SELF_SIGN_UP_REG_ENABLED = "EnableSignup";
-    public static final String SELF_SIGN_UP_REG_ROLE_NAME_ELEMENT = "RoleName";
-    public static final String SELF_SIGN_UP_REG_ROLE_IS_EXTERNAL = "IsExternalRole";
+    public static final String SELF_SIGN_UP_DEFAULT_CONFIG_FILE_PATH_OF_THE_CARBON_SUPER_USER = "/signupconfigurations/" +
```
Comment (tharindu1st) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIConstants.java:872]: unused ?

**Review by tharindu1st — APPROVED** [review]

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/utils/SelfSignUpUtil.java [hunk 2]
```diff
@@ -18,184 +18,31 @@
...
-				OMElement roleElement = (OMElement) roleListIterator.next();
-				String tmpRole = roleElement.getFirstChildWithName(
-						new QName(APIConstants.SELF_SIGN_UP_REG_ROLE_NAME_ELEMENT)).getText();
-				boolean tmpIsExternal = Boolean.parseBoolean(roleElement.getFirstChildWithName(
-						new QName(APIConstants.SELF_SIGN_UP_REG_ROLE_IS_EXTERNAL)).getText());
-				config.getRoles().put(tmpRole, tmpIsExternal);
-			}
-		} catch (XMLStreamException e) {
-			throw new APIManagementException("Error while parsing configuration ", e);
+		Object selfSighupConfigObject = ServiceReferenceHolder.getInstance().getApimConfigService()
```
Comment (RakhithaRR) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/utils/SelfSignUpUtil.java:40]: Typo in variable name

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/test/java/org/wso2/carbon/apimgt/impl/config/APIMConfigServiceImplTestCase.java [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+import org.wso2.carbon.apimgt.impl.utils.APIUtilTest;
+import java.io.File;
+import java.io.IOException;
+
+@RunWith(PowerMockRunner.class)
+@PrepareForTest({APIMConfigServiceImpl.class, SystemConfigurationsDAO.class})
+public class APIMConfigServiceImplTestCase {
+
+    @Test
+    public void testGetSelfSighupConfigWhenSelfSignUpIsEnabled() throws APIManagementException, IOException {
```
Comment (RakhithaRR) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/test/java/org/wso2/carbon/apimgt/impl/config/APIMConfigServiceImplTestCase.java:26]: Typo in the method name. Check other method names as well

Comment (thisaltennakoon) [reply components/apimgt/org.wso2.carbon.apimgt.impl/src/test/java/org/wso2/carbon/apimgt/impl/config/APIMConfigServiceImplTestCase.java:26]: Fixed in https://github.com/wso2/carbon-apimgt/pull/11919

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/utils/SelfSignUpUtil.java (hunk 2 above)
Comment (thisaltennakoon) [reply components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/utils/SelfSignUpUtil.java:40]: Fixed in https://github.com/wso2/carbon-apimgt/pull/11919

**Pull request merged**